zkevm-circuits/src/evm_circuit/util/math_gadget/rlp.rs

```diff
@@ -0,0 +1,419 @@
+use eth_types::{Address, Field, ToLittleEndian, ToScalar, Word};
+use gadgets::util::{and, expr_from_bytes, not, select, sum, Expr};
+use halo2_proofs::{
+    circuit::Value,
+    plonk::{Error, Expression},
+};
+
+use crate::evm_circuit::{
+    param::{N_BYTES_ACCOUNT_ADDRESS, N_BYTES_U64, N_BYTES_WORD},
+    util::{constraint_builder::ConstraintBuilder, CachedRegion, Cell, RandomLinearCombination},
+};
+
+use super::IsZeroGadget;
+
+#[derive(Clone, Debug)]
+pub struct RlpU64Gadget<F> {
+    /// Byte representation of the U64 value.
+    value_rlc: RandomLinearCombination<F, N_BYTES_U64>,
```

Comment:
I'm confused here. Why do we need randomness here? We don't need RLC to represent a `u64` using the RLP method anyway right?